src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.h

```diff
@@ -88,14 +92,40 @@ static inline struct drm_xocl_dev *bo_xocl_dev(const struct drm_xocl_bo *bo)
 	return bo->base.dev->dev_private;
 }
 
-static inline unsigned xocl_bo_ddr_idx(unsigned flags)
+static inline unsigned xocl_bo_ddr_idx(unsigned user_flags)
+{
+        return user_flags & 0xFFFFFF;
```

Comment:
please define a macro for user flag mask used for retrieving ddr idx and use ~macro to fetch bo type. make sure this macro is also shared with user land code when user flag is created.